packages/cojson/src/GarbageCollector.ts

```diff
@@ -0,0 +1,45 @@
+import { CoValueCore } from "./coValueCore/coValueCore.js";
+import { GARBAGE_COLLECTOR_CONFIG } from "./config.js";
+import { RawCoID } from "./ids.js";
+
+export class GarbageCollector {
+  private readonly coValueAccess = new Map<RawCoID, number>();
+  private readonly interval: ReturnType<typeof setInterval>;
+
+  constructor(private readonly coValues: Map<RawCoID, CoValueCore>) {
+    this.interval = setInterval(() => {
+      this.collect();
+    }, GARBAGE_COLLECTOR_CONFIG.INTERVAL);
+  }
+
+  getCurrentTime() {
+    return performance.now();
+  }
+
+  markCoValueAccess(coValue: CoValueCore) {
+    if (coValue.verified && coValue.verified.header.ruleset.type !== "group") {
+      this.coValueAccess.set(coValue.id, this.getCurrentTime());
+    }
+  }
+
+  collect() {
+    const currentTime = this.getCurrentTime();
+    for (const [id, accessTime] of this.coValueAccess) {
```

Comment:
Do you have a sense of how many covalues we'll be iterating over (like, order of magnitude)? I'm trying to get a sense of how long the garbage collection pause could take

Reply:
In the core server, without GC we go around 1 million of CoValues.

Can make the iteration allocation-free by storing the access timestamp as property on CoValueCore and iterating the coValues map values.

The operation should be fairly quick, will give you numbers, so we can also see if optimizing the iteration makes sense.

Also this is not a long-term solution, but more like a way to quickly introduce a garbage collection in order to see how it affects sync.

The final implementation will take into consideration more variables (such as the memory pressure and the size of the coValues) and will have to be more sophisticated.